### DISCUSSION TITLE
Firebase RTDB - Selective Delete (Delete with Query)

### DISCUSSION THREAD

**Discussion by tullius-ch** [SchneiderInfosystems/FB4D#64] [Q&A] [answered]
Hi

I am playing with Delphi 10.4.1 and FB4D. Even after hours of searching in the Web I did not found any example how to delete say the last 4 children of a node  or  deleting children with a timestamp that is older than a point in time. Even for me it is still not clear if this is possible in the Realtime Database.

Deleting a known node is no problem and reading all entries to "know" all nodes is some sort of overkill. 

The **fRealTimeDB.DeleteSynchronous(GetRTDBPath, Query)**  has a query parameter. Is there an example or a list what can be put into the query here? Some trials with a Query as in the GET gave me "unsupported operation"...

Thanks for any help.
Thomas

**Comment by ChristophSchneiderTrihow** [comment, marked as answer]
Hi Thomas
You should be able to use the filter options for _OrderBy_ and _LimitToFirst/Last_ in the Delete-Command as we do it for the Get-Command:
https://github.com/SchneiderInfosystems/FB4D/wiki/FB4D-Reference-IRealTimeDB#read-a-json-node

See also:
https://firebase.google.com/docs/database/rest/retrieve-data#section-rest-filtering 

But I have never tried that myself. Therefore I thank you for your feedback so that I can document this example also into the Wiki.
Greetings Christoph

**Reply by tullius-ch** [reply]
Hi Christoph

in between we had a bilateral chat about this topic but I want to close here with my findings so far. Probably it is helpful for someone else.
If I am not completely wrong it seems that actually the "selective delete" as described in the first chat is not supported. In my testprogram it always ends with an exception saying "Unsupported Operation" whereas a normal (unrestricted) delete works fine.
At the moment I delete like this: 1. selective Get 2. use the list to delete one by one.

**Reply by ChristophSchneiderTrihow** [reply]
Hi Thomas

Thank you for this update. It seems that the parameter _QueryParams_ is useless. I should remove it as consequence in the next bigger release. I will check this point.
